pkg/instrument/intrument.go

```diff
@@ -7,3 +7,9 @@ type CountRepoFunc func(store, repo, op string)
 
 // ObserveRepoFunc wraps a histogram to track repo op latencies.
 type ObserveRepoFunc func(store, repo, op string, begin time.Time)
+
+// CountRequestFunc wraps a counter to track number of received requests.
+type CountRequestFunc func(host, method, statusCode string)
+
+// ObserveRequestFunc wraps a histogram to track request latencies.
+type ObserveRequestFunc func(host, method, statusCode string, begin time.Time)
```

Comment:
Great follow-up on the instrument funcs. We can harden those a bit more and avoid having just strings as arguments. We know that `statusCode` is very vital to our telemetry as key driver for healthiness and is commonly expreseed as `int`. To encode this we could change the signature to:

``` go
type CountRequestFunc func(statusCode int, host, method, proto string)

type ObserveRequestFunc func(statusCode int, host, method, proto string, begin time.Time)
```

Then we leave it to the implementation of the funcs to convert the `statusCode` to something that can be used as label value (`string`). Also added `proto` which we should expect to be tracked.

Reply:
@xla  the problem with having `statusCode` of type int is that 
`func (h *Histogram) With(labelValues ...string) metrics.Histogram` requires all it's parameters to be strings. so a conversion must be made before the actual func is called

Reply:
Indeed label values are strings, what we can do tho is to shift the responsiblity of the conversion from the caller to the implementation. Where currently everywhere we call we need to convers before we can just make that part of the actual implementation of the func like:

``` go
type ObserveRequestFunc func(statusCode int, host, method, proto string, begin time.Time)

func requestMetrics() {
	requestLatencyFunc := func(statusCode int, host, method, proto string, begin time.Time) {
		// ...

		requestLatency.With(
			// ...
			labelStatusCode, strconv.Itoa(statusCode),
			// ...
		).Observe(time.Since(begin).Seconds())

		// ...
	}
}
```

Reply:
you're right. I just thought it would be ugly doing it this way but I like how it lends itself for tighter enforcement